### PR TITLE
feat: Add `Chat`

### PR DESCRIPTION
**Pull request by PoProstuMieciek**
# Added Chat

### Outside of task scope:
- fixed chat page height that was breaking the menu on the left

### CODE REVIEW

File: src/components/molecules/Message/Message.jsx [hunk 1]
```diff
@@ -11,16 +11,19 @@ export const MessageComponent = ({ children, date, isYours, profileID }) => {
...
-        userDetails.isSuccess && (
+    if (userDetails.data) {
+        const { profile_picture_url, displayName } =
+            userDetails.data.data() || {};
+
+        return (
             <div className={`message__field ${isYours ? 'ownMessage' : ''}`}>
                 <ImageComponent
-                    src={userDetails.data.data().profile_picture_url}
+                    src={profile_picture_url || ''}
```
Comment (T0ffik) [inline src/components/molecules/Message/Message.jsx:21]: Cyprian told me that if user have no profile picture we should display default picture

**Review by T0ffik — CHANGES_REQUESTED** [review]

File: src/components/molecules/Message/Message.jsx (hunk 1 above)
Comment (PoProstuMieciek) [reply src/components/molecules/Message/Message.jsx:21]: whats the path to the default profile picture?

**Review by Mariusz-web — APPROVED** [review]

**Comment by T0ffik** [comment]
I can't check od this works on phone so id someone will give me info about that I'm gonna approve

**Comment by PoProstuMieciek** [comment]
> I can't check od this works on phone so id someone will give me info about that I'm gonna approve

Trust me it works.

image

**Comment by Veryyapeee** [comment]
while entering the chat screen, we should be scrolled to the very bottom of the chat, currently we are staring at the very top

File: src/components/molecules/Message/Message.jsx [hunk 2]
```diff
@@ -29,8 +33,10 @@ export const MessageComponent = ({ children, date, isYours, profileID }) => {
                     {children}
                 </Text>
             </div>
-        )
-    );
+        );
+    }
+
+    return null;
```
Comment (Veryyapeee) [inline src/components/molecules/Message/Message.jsx:39]: returning null should be first, more clean code, I mean replace that with
if(!userDetails.data){
return null
}

File: src/components/organisms/MessageCollection/MessageCollection.jsx [hunk 3]
```diff
@@ -1,23 +1,37 @@
...
  * @param {{ userMsgs: import('../../../api/types').Message[] }} param0
  */
 export const MessageCollection = ({ userMsgs }) => {
-    const user = {};
+    const user = useUser();
+
+    const messageEl = useRef(null);
+    useEffect(() => {
+        if (messageEl) {
+            messageEl.current.addEventListener('DOMNodeInserted', (event) => {
```
Comment (Veryyapeee) [inline src/components/organisms/MessageCollection/MessageCollection.jsx:15]: remove eventListener when component unmounts

File: src/pages/Chat/Chat.jsx [hunk 4]
```diff
@@ -1,14 +1,55 @@
...
     const params = useParams();
     const chatID = id || params.id;
 
-    return <Text>{chatID}</Text>;
+    const user = useUser();
+    const chatRoom = useChatRoom(chatID);
+    const messages = useMessages(chatID);
+    const createMessage = useCreateMessage(chatID);
+
+    if (chatRoom.isLoading || messages.isLoading) return null;
```
Comment (Veryyapeee) [inline src/pages/Chat/Chat.jsx:24]: if(){}

**Review by Veryyapeee — CHANGES_REQUESTED** [review]

**Review by Veryyapeee — APPROVED** [review]

**Pull request merged**